kadi/commands/tests/test_validate.py

```diff
@@ -1,67 +1,105 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+"""Test the validation of command loads.
+
+This relies on regression data files that are generated or updated by running this test
+file with the environment variable KADI_VALIDATE_WRITE_REGRESSION set::
+
+  $ env KADI_VALIDATE_WRITE_REGRESSION=1 pytest -k test_validate -s -v
+
+Regression testing uses a 5-day period covering a safe mode with plenty of things
+happening. There are a number of violations in this period and a couple of excluded
+intervals.
+"""
+
 import functools
-import gzip
-import pickle
+import os
 from pathlib import Path
 
 import numpy as np
 import pytest
 import ska_sun
+import yaml
+from cxotime import CxoTime
+from ska_helpers.utils import temp_env_var
 from testr.test_helper import has_internet
 
+import kadi.commands as kc
 from kadi.commands.utils import compress_time_series
 from kadi.commands.validate import (
     Validate,
     ValidateRoll,
 )
 
-# Regression testing for this 5-day period covering a safe mode with plenty of things
-# happening. There are a number of violations in this period and a couple of excluded
-# intervals.
 REGRESSION_STOP = "2022:297"
 REGRESSION_DAYS = 5
 
 HAS_INTERNET = has_internet()
 
 
-def write_regression_data(stop, days, no_exclude):
-    cwd = Path(__file__).parent
+@pytest.fixture(scope="module")
+def local_testing(tmp_path_factory):
+    """Several context managers to set up the environment for local testing."""
+    kc.clear_caches()
+    cmds_dir = tmp_path_factory.mktemp("cmds_dir")
+    with (
+        kc.conf.set_temp("cache_loads_in_astropy_cache", True),
+        kc.conf.set_temp("clean_loads_dir", False),
+        kc.conf.set_temp("commands_dir", str(cmds_dir)),
+    ):
+        yield
+    kc.clear_caches()
+
+
+@pytest.fixture()
+def regress_stop(local_testing):
+    """Set the default stop time for regression testing
+
+    This will force regenerating "recent" commands in case command events or command
+    sets have changed.
 
-    print(f"Getting validation regression data for {stop} {days} {no_exclude}")
-    data_all = get_all_validator_data(stop, days, no_exclude)
-    name = f"validators_{stop.replace(':', '')}_{days}_{no_exclude}.pkl.gz"
-    path = cwd / "data" / name
+    This fixture cannot be combined with the module-scoped ``local_testing`` fixture
+    because this causes problems for other tests that rely on the default stop time.
```

Comment:
I'm a little confused here, as with the signature `regress_stop(local_testing)` isn't the fixture already combined with the local_testing fixture?

Reply:
Sorry, the comment is indeed a little confusing. It should be "**The commands in** this fixture cannot be included in the module-scoped ...". Basically setting `KADI_COMMANDS_DEFAULT_STOP` for the entire module will break some tests.